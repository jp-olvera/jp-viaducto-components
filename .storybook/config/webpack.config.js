const path = require('path');
module.exports = {
  entry: './src/index.js',
  output: {
    path: path.resolve(__dirname, '../dist'),
    filename: 'index.js',
    libraryTarget: 'commonjs2',
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        exclude: /(node_modules|build|dist)/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: ['@babel/preset-env', '@babel/preset-react'],
          },
        },
      },
      {
        test: /\.css$/,
        exclude: /(node_modules|build)/,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: /\.svg$/,
        exclude: /(node_modules|build)/,
        use: ['url-loader'],
      },
    ],
  },
  externals: {
    react: {
      commonjs: 'react',
      commonjs2: 'react',
      amd: 'React',
      root: 'React',
    },
    'prop-types': {
      commonjs: 'prop-types',
      commonjs2: 'prop-types',
      amd: 'PropTypes',
      root: 'PropTypes',
    },
  },
  resolve: {
    extensions: ['.js', '.jsx', '.css'],
    alias: {
      react: path.resolve(__dirname, './node_modules/react'),
      'prop-types': path.resolve(__dirname, './node_modules/prop-types'),
      'styled-components': path.resolve(
        __dirname,
        '../node_modules/styled-components'
      ),
    },
  },
};
