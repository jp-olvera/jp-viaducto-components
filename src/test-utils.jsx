import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
