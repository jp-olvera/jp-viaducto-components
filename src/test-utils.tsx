import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
