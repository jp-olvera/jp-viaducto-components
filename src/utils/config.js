const config = {
  spacing: {
    none: '0px',
    nano: '0.279rem',
    micro: '0.335rem',
    tiny: '0.402rem',
    xs: '0.482rem',
    sm: '0.694rem',
    md: '1.2rem',
    lg: '2.074rem',
    xl: '2.488rem',
    xxl: '2.986rem',
  },
  display: {
    mobile: {
      xs: '2.986rem',
      sm: '3.012rem',
      md: '3.213rem',
      lg: '3.658rem',
      xl: '4.165rem',
      xxl: '5.06rem',
    },
    desktop: {
      xs: '3.583rem',
      sm: '5.16rem',
      md: '6.192rem',
      lg: '8.916rem',
      xl: '12.839rem',
      xxl: '15.407rem',
    },
  },
  breakpoints: {
    xs: '20rem', //'320px'
    sm: '36rem', //'576px'
    md: '48rem', //'768px'
    lg: '62rem', //'992px'
    xl: '90rem', //'1440px'
  },
  colors: {
    primary: {
      default: '#937B3D',
      hover: '#AD9043',
      click: '#C3A24A',
    },
    secondary: {
      default: '#573D3D',
      hover: '#744D4D',
      click: '#8A5E5E',
    },
    info: {
      default: '#75CDFF',
      hover: '#90D7FF',
      click: '#D9F1FF',
    },
    success: {
      default: '#31A74B',
      hover: '#2FBD4E',
      click: '#3AE25F',
    },
    warning: {
      default: '#FFDF38',
      hover: '#FFEA7C',
      click: '#FFF1A5',
    },
    danger: {
      default: '#FF0000',
      hover: '#FF5454',
      click: '#FF8686',
    },
  },
  text: {
    white: '#FFFFFF',
    lightGray: '#EFEFEF',
    mutedGray: '#A0A0A0',
    darkGray: '#5A5A5A',
    dark: '#050505',
    primary: '#937B3D',
    secondary: '#573D3D',
    info: '#75CDFF',
    success: '#31A74B',
    warning: '#FFDF38',
    danger: '#FF0000',
  },
};

export default config;
