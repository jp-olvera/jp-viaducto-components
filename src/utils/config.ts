import { ConfigProps } from 'ballena-types';

const config: ConfigProps = {
  breakpoints: {
    xs: '20rem', // '320px'
    sm: '36rem', // '576px'
    md: '48rem', // '768px'
    lg: '62rem', // '992px'
    xl: '90rem', // '1440px'
  },
  border: '0.063rem solid #001D48',
  colors: {
    background: '#fff',
    primary: {
      default: '#937B3D',
      hover: '#AD9043',
      click: '#C3A24A',
      text: '#fff',
      shadow: 'rgba(147, 123, 62, 0.53)',
    },
    secondary: {
      default: '#573D3D',
      hover: '#744D4D',
      click: '#8A5E5E',
      text: 'white',
      shadow: 'rgba(87, 61, 61, 0.53)',
    },
    info: {
      default: '#75CDFF',
      hover: '#90D7FF',
      click: '#D9F1FF',
      text: '#000',
      shadow: 'rgba(117, 205, 255, 0.53)',
    },
    success: {
      default: '#31A74B',
      hover: '#2FBD4E',
      click: '#3AE25F',
      text: '#000',
      shadow: 'rgba(49, 167, 75, 0.53)',
    },
    warning: {
      default: '#FFDF38',
      hover: '#FFEA7C',
      click: '#FFF1A5',
      text: '#000',
      shadow: 'rgba(255, 223, 56, 0.53)',
    },
    danger: {
      default: '#FF0000',
      hover: '#FF5454',
      click: '#FF8686',
      text: 'white',
      shadow: 'rgba(255, 0, 0, 0.53)',
    },
    tab: {
      default: '#F1F1F1',
      click: '#4F83CC',
      hover: '#01579B',
      text: '#000',
      shadow: 'rgba(1, 87, 155, 0.53)',
    },
    text: {
      danger: '#B71C1C',
      dangerLight: '#F05545',
      dangerDark: '#7F0000',
      dark: '#050505',
      darkGray: '#5A5A5A',
      info: '#64B5F6',
      infoLight: '#9BE7FF',
      infoDark: '#2286C3',
      lightGray: '#EFEFEF',
      mutedGray: '#A0A0A0',
      primary: '#937B3D',
      secondary: '#573D3D',
      success: '#689F38',
      successLight: '#99D066',
      successDark: '#387002',
      warning: '#FDD835',
      warningLight: '#FFFF6B',
      warningDark: '#C6A700',
      white: '#FFFFFF',
    },
    defaultInputBorderColor: '#001D48',
    disableColor: '#BFBFBF',
    neutral: {
      light: '#FFD1D1',
      invert: '#000',
    },
    surface: {
      light: '#F9F9F9',
      dark: '#242020',
    },
    table: {
      background: '#fff',
      footerColor: '#fff',
      headerColor: '#FAFAFA',
      selectedColor: '#d9d9d9',
      zebraColor: '#F6F8FA',
      hoverColor: '#D1D5DA',
    },
    iconColor: '#595959',
  },
  controlHeight: {
    xsmall: '1.2rem',
    small: '2.074rem',
    default: '2.488rem',
    large: '2.986rem',
  },
  drawerSizes: {
    sm: '20rem',
    md: '53rem',
    lg: '64rem',
  },
  display: {
    mobile: {
      xs: '2.986rem',
      sm: '3.012rem',
      md: '3.213rem',
      lg: '3.658rem',
      xl: '4.165rem',
      xxl: '5.06rem',
    },
    desktop: {
      xs: '3.583rem',
      sm: '5.16rem',
      md: '6.192rem',
      lg: '8.916rem',
      xl: '12.839rem',
      xxl: '15.407rem',
    },
  },
  radius: {
    none: '0rem',
    sm: '0.188rem',
    md: '0.375rem',
    lg: '0.75rem',
  },
  spacing: {
    none: '0rem',
    nano: '0.279rem',
    micro: '0.335rem',
    tiny: '0.402rem',
    xs: '0.482rem',
    sm: '0.694rem',
    md: '1.2rem',
    lg: '2.074rem',
    xl: '2.488rem',
    xxl: '2.986rem',
    xxxl: '3.5rem',
  },
  toasterPlacement: 'top-right',
  fontFamily: 'Arial',
  titleFontFamily: 'system-ui',
};

export default config;
