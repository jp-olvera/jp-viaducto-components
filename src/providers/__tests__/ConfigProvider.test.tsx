/* eslint-env jest */

import React, { useContext, useEffect } from 'react';
import '@testing-library/jest-dom/extend-expect';
import styled from 'styled-components';
import config from '../../utils/config';
import { render } from '../../test-utils';
import { ConfigProvider, ConfigContext } from '..';

describe('<ConfigProvider/>', () => {
  test('should create snapshot of ConfigContext', () => {
    const updateConfig = jest.fn();
    const conf: any = {};
    const { container } = render(
      <ConfigContext.Provider value={{ configuration: conf, updateConfig }}>
        <div>AAAAA</div>
      </ConfigContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should create snapshot ConfigProvider', () => {
    const setState = jest
      .fn()
      .mockImplementation(() => ({ spacing: { hola: '1rem' } }));
    const useStateMock: any = jest.spyOn(React, 'useState');
    useStateMock.mockImplementation((init: any) => [
      {
        ...init,
        ...{ spacing: { hola: 1 }, toasterPlacement: 'centerFailed' },
      },
      setState,
    ]);
    const { container } = render(
      <ConfigProvider>
        <div>AAAAA</div>
      </ConfigProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should change config', () => {
    const MyComponent = styled.div < any > `
      color: red;
    `;
    const A = () => {
      const { updateConfig } = useContext(ConfigContext);
      const myConfig = {
        ...config,
        breakpoints: {
          // for the media queries
          xs: '20rem', // '320px'
          sm: '36rem', // '576px'
          md: '48rem', // '768px'
          lg: '62rem', // '992px'
          xl: '90rem', // '1440px'
          // ... or any sizes
        },
      };
      useEffect(() => {
        updateConfig(myConfig);
      }, []);

      return (
        <ConfigProvider>
          <MyComponent configuration={myConfig}>Hola</MyComponent>
        </ConfigProvider>
      );
    };

    const { container } = render(<A />);
    expect(container).toMatchSnapshot();
  });
});
