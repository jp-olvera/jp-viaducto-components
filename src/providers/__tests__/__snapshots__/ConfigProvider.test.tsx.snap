// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigProvider/> should change config 1`] = `
<div>
  <div
    class="ConfigProvidertest__MyComponent-sc-153tyaj-0 hkxlih"
  >
    Hola
  </div>
</div>
`;

exports[`<ConfigProvider/> should create snapshot ConfigProvider 1`] = `
<div>
  <div>
    AAAAA
  </div>
</div>
`;

exports[`<ConfigProvider/> should create snapshot of ConfigContext 1`] = `
<div>
  <div>
    AAAAA
  </div>
</div>
`;
